patches/server/0970-New-specific-event-watching-command.patch

```diff
@@ -0,0 +1,220 @@
+diff --git a/src/main/java/io/papermc/paper/command/subcommands/EventVerboseCommand.java b/src/main/java/io/papermc/paper/command/subcommands/EventVerboseCommand.java
+index 165bb5324aa06e4bd204a570b45b185b0bd11a84..f4ad22037f0f7dfac5e8bf357a27f1f184c3cf3f 100644
+--- a/src/main/java/io/papermc/paper/command/subcommands/EventVerboseCommand.java
++++ b/src/main/java/io/papermc/paper/command/subcommands/EventVerboseCommand.java
+@@ -127,11 +127,11 @@ public class EventVerboseCommand implements PaperSubcommand {
+             this.event = event;
+         }
+ 
+-        @Nullable
+         /**
+          * Properly serialize our Wrapper into a JsonObject, so we can log it properly
+          * @implNote Is null if the event has only final fields
+          */
++        @Nullable
+         private JsonObject serialize() {
+             JsonObject parent = new JsonObject();
+             parent.addProperty("timestamp", timestamp);
+diff --git a/src/main/java/io/papermc/paper/command/subcommands/EventWatcherCommand.java b/src/main/java/io/papermc/paper/command/subcommands/EventWatcherCommand.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..323596f65198f53c8c6cec200ee37723b78fb8f4
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/command/subcommands/EventWatcherCommand.java
+@@ -0,0 +1,151 @@
++package io.papermc.paper.command.subcommands;
++
++import com.google.common.collect.Lists;
++import com.google.gson.JsonArray;
++import com.google.gson.JsonObject;
++import com.google.gson.internal.Streams;
++import com.google.gson.stream.JsonWriter;
++import io.papermc.paper.command.PaperSubcommand;
++import net.minecraft.server.MinecraftServer;
++import org.bukkit.command.CommandSender;
++import org.bukkit.event.Event;
++import org.bukkit.event.Listener;
++import org.checkerframework.checker.nullness.qual.NonNull;
++import org.checkerframework.checker.nullness.qual.Nullable;
++import org.checkerframework.framework.qual.DefaultQualifier;
++
++import java.io.IOException;
++import java.io.PrintStream;
++import java.io.StringWriter;
++import java.lang.reflect.Field;
++import java.nio.charset.StandardCharsets;
++import java.nio.file.Files;
++import java.nio.file.Path;
++import java.time.LocalDateTime;
++import java.time.format.DateTimeFormatter;
++import java.util.*;
++
++import static net.kyori.adventure.text.Component.text;
++import static net.kyori.adventure.text.format.NamedTextColor.GREEN;
++import static net.kyori.adventure.text.format.NamedTextColor.RED;
++
++@DefaultQualifier(NonNull.class)
++public class EventWatcherCommand implements PaperSubcommand {
++
++    /**
++     * The key is the equivalent of {@link Class#getSimpleName()} on the class,
++     * and value is the file we're trying to log to
++     */
++    private static final List<Watcher> watching = Collections.synchronizedList(new ArrayList<>()); //only sync in the rare case that multiple people are debugging, should be unnecessary
```

Comment:
This should be a `Map<Class<? extends Event>, smth>` instead of a List. twice you are doing streams to filter to find the one that matches the supplied class.